### ISSUE TITLE
add event to existing ics file?

### ISSUE THREAD

**Issue by tomaslundborg** [derhuerst/ics-service#3]
I want to offer this to clients. And, once they make a booking, the new event is added to the _already existing_ ics file.

Can only see that one can generate new, but what about parsing existing and adding event to that, saving it again? And waiting for the outlook/google cal etc to poll new updates. 

Or am I thinking about this in the wrong way? How do ppl usually do this?

**Comment by derhuerst** [comment]
First a general note: This package *does not* assume you're storing a regular `.ics` file somewhere. It is intended for serving a "calendar feed", which is an iCalendar file served via HTTP.

I'm going to assume you want to extend an existing calendar feed with another event (or set of events). Generating a separate new feed is not an option?

**Comment by henkvantijen** [comment]
I am searching for a calendar feed that one can subscribe to, but that also pushes changes and additions to the subscribers too. As I understand, webcal://  is the (a?) protocol for this.  

https://stackoverflow.com/questions/23206761/exposing-calendar-as-an-ical-with-subscribe-option-in-php

https://en.wikipedia.org/wiki/Webcal

My municipal cleaner service publishes the garbage pick up dates a year in advance.  But if there are any changes, I see those in my Google calendar.  This is the uri for my  street
webcal://wasteprod2api.ximmio.com/api/CallIcal?cn=Avalex&x=f7a74ad1-fdbf-4a43-9f91-44644f4d4222&ty=&ua=571700cc4a1157156d2a50fbaff49fb62441c767&sd=2022-12-25&ed=2024-01-27&path=https://wasteprod2api.ximmio.com&ln=nl&nt=7

Is this possible by ics service?  If not, would it be a big effort to add ?

Kind regards.

**Comment by derhuerst** [comment]
> I am searching for a calendar feed that one can subscribe to, but that also pushes changes and additions to the subscribers too. As I understand, webcal:// is the (a?) protocol for this.

AFAIK pushing to users' devices is not possible with any of the calendar feed/subscription formats. All of them assume that the client (or some cloud service acting on behalf of the client) polls the calendar feed regularly.

The iCal/`webcal:` ecosystem is not very well-specified AFAICT. It's a lot of de-facto standards and some actual formal standards put together, with implementations slightly varying in how they work. Nevertheless, a basic calendar feed should work with all major calendar apps/services.

> Is this possible by ics service? If not, would it be a big effort to add ?

Yes, exactly this is `ics-service`'s goal: You specify calendar events in the [`ics@3`](https://github.com/adamgibbons/ics/tree/v3.0.0) format, `ics-service` serves a calendar feed containing them.

**Comment by derhuerst** [comment]
Closing this. Please re-open if you have further related questions.

TLDR: Yes, it would be possible to build such an augmenting calendar feed. `ics-service` is more lower-level though, so it doesn't have this functionality built-in.
